### PR TITLE
PWA-2387: Replace Magento CE Homepage with Page Builder

### PR DESCRIPTION
**Pull request by michaelyu0123**
- Deprecated CategoryList component
- Updated homepage to default to using pagebuilder
- Updated fallback messaging
- Updated test coverage

## Description
PWA Studio has two separate "homepage" contents: one for when Page Builder is present, and one when it isn't.

This was necessary because Page Builder used to be Adobe Commerce (formerly called "Enterprise Edition") only. Now that Page Builder is open source, the Magento Open Source (formerly called "Community Edition") version of our demo storefronts can confidently rely on Page Builder content being present.

## Related Issue

Closes [PWA-2387](https://jira.corp.magento.com/browse/PWA-2387)

## Acceptance

1. Venia homepage shows Page Builder content
2. The CategoryList component is marked as deprecated
3. When Page Builder CMS page content is not available, a fallback message appears instead

## Verification Steps

Preconditions:

1. Page Builder delivers CMS page content
2. PWA Studio is built / running with MAGENTO_BACKEND_EDITION=CE

Test Plan:

1. Deploy (or run locally) an instance of Venia with MAGENTO_BACKEND_EDITION=CE
2. Verify that the homepage shows PageBuilder CMS content
3. Verify the same when MAGENTO_BACKEND_EDITION=EE
4. Update backing instance Page Builder CMS content to not return anything (empty)
5. Verify that a fallback message is displayed instead

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### CODE REVIEW

File: packages/peregrine/lib/talons/Cms/useCmsPage.js [hunk 1]
```diff
@@ -52,22 +52,9 @@ export const useCmsPage = props => {
...
-            cmsPage.content.length > 0 &&
-            !cmsPage.content.includes('CMS homepage content goes here.')
-        );
-    }, [cmsPage]);
-
     return {
         cmsPage,
         error,
-        hasContent,
         rootCategoryId,
```
Comment (mikhaelbois) [inline packages/peregrine/lib/talons/Cms/useCmsPage.js:58]: `error` and `rootCategoryId` does not seem like they are used anymore.

**Review by mikhaelbois — CHANGES_REQUESTED** [review]

File: packages/peregrine/lib/talons/Cms/useCmsPage.js (hunk 1 above)
Comment (michaelyu0123) [reply packages/peregrine/lib/talons/Cms/useCmsPage.js]: removed them.

**Review by mikhaelbois — APPROVED** [review]
LGTM ✅

**Comment by dpatil-magento** [comment]
QA Approved.

**Pull request merged**